service.watchedlist/resources/lib/utils.py

```diff
@@ -0,0 +1,243 @@
+"""
+This file contains additional utility functions
+"""
+import xbmc, xbmcgui, xbmcvfs, xbmcaddon
+import os
+import time
+import sys
+import buggalo
+import re
+import random
+
+__addon_id__= u'service.watchedlist'
+__Addon = xbmcaddon.Addon(__addon_id__)
+
+# XBMC-JSON
+if sys.version_info < (2, 7):
+    import simplejson
+else:
+    import json as simplejson
+
+
+def data_dir():
+    """"get user data directory of this addon. 
+    according to http://wiki.xbmc.org/index.php?title=Add-on_Rules#Requirements_for_scripts_and_plugins
+    """
+    __datapath__ = xbmc.translatePath( __Addon.getAddonInfo('profile') ).decode('utf-8')
+    if not xbmcvfs.exists(__datapath__):
+        xbmcvfs.mkdir(__datapath__)
+    return __datapath__
+
+def addon_dir():
+    """"get source directory of this addon.
+    according to http://wiki.xbmc.org/index.php?title=Add-on_Rules#Requirements_for_scripts_and_plugins
+    """
+    return __Addon.getAddonInfo('path').decode('utf-8')
+
+def log(message,loglevel=xbmc.LOGNOTICE):
+    """"save message to xbmc.log.
+    
+    Args:
+        message: has to be unicode, http://wiki.xbmc.org/index.php?title=Add-on_unicode_paths#Logging
+        loglevel: xbmc.LOGDEBUG, xbmc.LOGINFO, xbmc.LOGNOTICE, xbmc.LOGWARNING, xbmc.LOGERROR, xbmc.LOGFATAL
+    """
+    xbmc.log(encode(__addon_id__ + u": " + message), level=loglevel)
+
+
+def showNotification(title,message, time=4000):
+    """Show Notification
+
+    Args: 
+        title: has to be unicode
+        message: has to be unicode
+        time: Time that the message is beeing displayed
+    """
+    __addoniconpath__ = os.path.join(addon_dir(),"icon.png")
+    log(u'Notification. %s: %s' % (title, message) )
+    if xbmc.Player().isPlaying() == False: # do not show the notification, if a video is being played.
+        xbmc.executebuiltin(encode('Notification("' + title + '","' + message + '",'+(str(time)).decode('utf-8')+',"' + __addoniconpath__ + '")'))
```

Comment:
is is preferred to use xbmcgui.Dialog().Notification() instead of calling a builtin.